src/NServiceBus.Extensions.DispatchRetries/Behaviors/BatchDispatchRetriesBehavior.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Threading.Tasks;
+using NServiceBus.Pipeline;
+using NServiceBus.Settings;
+using Polly;
+
+namespace NServiceBus.Extensions.DispatchRetries.Behaviors
+{
+    public class BatchDispatchRetriesBehavior : Behavior<IBatchDispatchContext>
+    {
+        private readonly ReadOnlySettings _readOnlySettings;
+
+        public BatchDispatchRetriesBehavior(ReadOnlySettings readOnlySettings)
+        {
+            _readOnlySettings = readOnlySettings;
+        }
+
+        public override Task Invoke(IBatchDispatchContext context, Func<Task> next)
+        {
+            if (context.Extensions.TryGet(Constants.BatchDispatchRetryPolicy, out AsyncPolicy retryPolicy))
+            {
+                return retryPolicy.ExecuteAsync(next);
+            }
+
+            if (_readOnlySettings.TryGet(Constants.DefaultBatchDispatchRetryPolicy, out AsyncPolicy defaultRetryPolicy))
```

Comment:
You could extract the default policy once and then reuse so that you don't always have to get it again during execution